storage/uri.go

```diff
@@ -118,8 +142,70 @@ func parentGeneric(location string) (string, error) {
 	return parent, nil
 }
 
-// Parent gets the parent of a URI by splitting it along '/' separators and
-// removing the last item.
+// NewURI creates a new URI from the given string representation. This could be
+// a URI from an external source or one saved from URI.String()
+//
+// Deprecated - use ParseURI instead
+func NewURI(s string) fyne.URI {
+	u, _ := ParseURI(s)
+	return u
+}
+
+// ParseURI creates a new URI instance by parsing a URI string, which must
+// conform to IETF RFC3986.
+//
+// Since 2.0.0
+func ParseURI(s string) (fyne.URI, error) {
+
+	if len(s) > 5 && s[:5] == "file:" {
+		path := s[5:]
+		if len(path) > 2 && path[:2] == "//" {
+			path = path[2:]
+		}
+
+		// this looks weird, but it makes sure that we still pass
+		// url.Parse()
+		s = NewFileURI(path).String()
+	}
+
+	_, err := url.Parse(s)
```

Comment:
Why not store the result in the struct and avoid all the `// NOTE: we verified in ParseURI() that this would not error.` above?

Reply:
66f8334becd5e190d8e03d59264403fd18d2c515